packages/provider/cloudscale.yml

```diff
@@ -0,0 +1,36 @@
+classes:
+  - .common
+
+parameters:
+
+  pkg.provider:
+    cloudscale:
+      version: v0.3.0
+
+  crossplane:
+    providers:
+      cloudscale:
+        package: ghcr.io/vshn/provider-cloudscale/provider:${pkg.provider:cloudscale:version}
+    providerConfigs:
+      cloudscale-lpg-2:
+        apiVersion: cloudscale.crossplane.io/v1
+        spec:
+          credentials:
+            source: InjectedIdentity
+            apiTokenSecretRef:
+              name: cloudscale-lpg-2
+              # This is not resolving when during "commodore package compile"
+              namespace: ${crossplane:namespace}
+
+  appcat:
+    secrets:
+      cloudscale-lpg-2:
+        metadata:
+          namespace: ${crossplane:namespace}
+        stringData:
+          CLOUDSCALE_API_TOKEN: '?{vaultkv:${cluster:tenant}/${cluster:name}/appcat/objectstorage/provider-cloudscale/cloudscale-lpg-2}'
```

Comment:
I went with explicitly defining our clusters here, with the intention that some point we will have cross-cluster and cross-cloud-provider configs, where users can deploy buckets in other cloud providers by switching out the composition.

Reply:
On second thought, this is actually not required.

For example: If there's `cloudscale-rma-1` cluster, then the API token should point to a project like `appcat-objects-cloudscale-rma-1`. Also, since in the resources (at least for S3) users can still choose a different bucket region. Yet that bucket and objectsuser should remain in the `appcat-objects-cloudscale-rma-1` project, even if the bucket ends up in `lpg`.

For cross-provider-usage: Let's assume we have a cluster `exoscale-ch-gva-2` and we want to support creating buckets on cloudscale's `lpg` region (e.g. for offsite-backup buckets). This time we create a cloudscale Project named `appcat-objects-exoscale-ch-gva-2` and it contains the objects users and bucket of `exoscale-ch-gva-2`. There's no link to `cloudscale-lpg-2` in any way.